netplan/cli/commands/apply.py

```diff
@@ -290,17 +295,25 @@ def command_apply(self, run_generate=True, sync=False, exit_on_error=True, state
                 utils.ip_addr_flush(iface)
             # clear NM state, especially the [device].managed=true config, as that might have been
             # re-set via an udev rule setting "NM_UNMANAGED=1"
-            subprocess.check_call(['rm', '-rf', '/run/NetworkManager/devices'])
+            subprocess.check_call(['rm', '-rf', '/run/NetworkManager/devices'], env=env)
             utils.systemctl_network_manager('start', sync=sync)
             if sync:
-                # wait up to 2 sec for 'STATE=connected (site/local-only)' or
-                # 'STATE=connected' to appear in 'nmcli general' STATE
-                env = dict(os.environ, LC_ALL='C')
+                # 'nmcli' could be /usr/bin/nmcli or
+                # /snap/bin/nmcli -> /snap/bin/network-manager.nmcli
                 cmd = ['nmcli', 'general', 'status']
-                for _ in range(20):
-                    if b'\nconnected' in subprocess.check_output(cmd, env=env):
+                # wait a bit for 'connected (site/local-only)' or
+                # 'connected' to appear in 'nmcli general' STATE
+                for _ in range(10):
+                    out = subprocess.run(cmd, capture_output=True,
+                                         universal_newlines=True, env=env)
```

Comment:
Up to you to use it here, not use it, or use it everywhere ; but it does improve readability IMO:

> Changed in version 3.7: Added the text parameter, as a more understandable alias of universal_newlines.
https://docs.python.org/3/library/subprocess.html#subprocess.run

Reply:
ACK. I've been holding back on this one consciously, for backwards compatibility reasons. But we're mostly targeting Python 3.10 (Jammy) nowadays, so should be fine to do this refactoring.

It should be done in an independent PR, though.